modules/light-clients/06-solomachine/update.go

```diff
@@ -44,7 +44,25 @@ func (cs ClientState) verifyHeader(ctx sdk.Context, cdc codec.BinaryCodec, clien
 	}
 
 	// assert currently registered public key signed over the new public key with correct sequence
-	data, err := HeaderSignBytes(cdc, header)
+	headerData := &HeaderData{
+		NewPubKey:      header.NewPublicKey,
+		NewDiversifier: header.NewDiversifier,
+	}
+
+	dataBz, err := cdc.Marshal(headerData)
+	if err != nil {
+		return err
+	}
+
+	signBytes := &SignBytes{
+		Sequence:    header.Sequence,
+		Timestamp:   header.Timestamp,
+		Diversifier: header.NewDiversifier,
```

Comment:
Interesting, I think this is a :bug: Shouldn't the old public key sign over the old divisifier and not the new :eyes:

Reply:
Yes, I think you're right! So this should be verifying that the current public key with current diversifier has signed over the data containing the new header details (new pubKey, new diversifier)? 

I can push an update to change this to `cs.ConsensusState.Diversifier`